JS/Lesson34/goksel/recipe-app/src/modules/recipes/NewRecipe/NewRecipeForm/index.jsx

```diff
@@ -0,0 +1,233 @@
+import { useForm } from "react-hook-form";
+import { MealCategory } from "./newRecipeHelper";
+import { MealArea } from "./newRecipeHelper";
+import { handlestrIngredientsAndMeasures } from "./newRecipeHelper"
+import { Button, FormControl, Grid, InputLabel, MenuItem, Select, Typography } from '@mui/material';
+import TextField from '@mui/material/TextField';
+
+
+export const NewRecipe = () => {
+
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+        reset
+    } = useForm();
+
+    const onSubmit = (data) => {
+
+        const dateObject = new Date();
+        const formattedDate = dateObject.toISOString();
+
+        const ingredientsAndMeasures = handlestrIngredientsAndMeasures(data.strIngredientsAndMeasures);
+        delete data.strIngredientsAndMeasures;
+        const cleanedData = {
+            ...data,
+            strMeal: data.strMeal.trim().replace(/\s+/g, " "),
+            strDrinkAlternate: data.strDrinkAlternate.trim().replace(/\s+/g, " "),
+            strInstructions: data.strInstructions.trim().replace(/\s+/g, " "),
+            strTags: data.strTags.trim().replace(/\s+/g, ""),
+            strYoutube: data.strYoutube.trim().replace(/\s+/g, " "),
+            dateModified: formattedDate,
+            ...ingredientsAndMeasures
+        }
+
+        console.log(cleanedData);
+        reset();
+    }
+
+    return (
+        <Grid
+            container
+            component="form"
+            onSubmit={handleSubmit(onSubmit)}
+            noValidate
+            spacing={2}
+            direction="column"
+            alignItems="center"
+            sx={{mt:10}}
+        >
+
+            <Grid item>
+                <TextField
+                    sx={{ minWidth: {xs:300, sm:400, md: 500, lg:600}}}
```

Comment:
'sx={{ minWidth: {xs:300, sm:400, md: 500, lg:600}}}'is repeated everywhere. Save it to a variable before 'return' statement and use it as a variable instead.

Reply:
A better approach would be to restrict the width of the grid item (parent element) depending on the screen size, then you would not need to duplicate the code fore each input field.